src/test/java/com/exasol/adapter/jdbc/BaseTableMetadataReaderTest.java

```diff
@@ -103,15 +105,61 @@ void testWarnIfTableScanResultsAreEmpty() throws SQLException {
 
     @Test
     void testValidateMappedTablesListSize() throws SQLException {
+        final ResultSet resultSetMock = resultSetWithUnlimitedSize();
+        when(this.columnMetadataReaderMock.mapColumns("table"))
+                .thenReturn(List.of(ColumnMetadata.builder().name("column").type(DataType.createBool()).build()));
+        final TableMetadataReader metadataReader = createDefaultTableMetadataReader();
+        final List<String> noFilter = Collections.emptyList();
+        final RemoteMetadataReaderException exception = assertThrows(RemoteMetadataReaderException.class,
+                () -> metadataReader.mapTables(resultSetMock, noFilter));
+        assertAll( //
+                () -> assertThat(exception.getMessage(), containsString("E-VSCJDBC-42")), //
+                () -> assertThat(exception.getMessage(), containsString("1000")) //
+        );
+    }
+
+    private ResultSet resultSetWithUnlimitedSize() throws SQLException {
+        final ResultSet resultSet = Mockito.mock(ResultSet.class);
+        when(resultSet.next()).thenReturn(true);
+        when(resultSet.getString(NAME_COLUMN)).thenReturn("table");
+        return resultSet;
+    }
+
+    // verify that the actual table limit is part of the error message
+    @Test
+    void testValidateMappedTablesListSizeWithProperty2000() throws SQLException {
         final ResultSet resultSetMock = Mockito.mock(ResultSet.class);
         when(resultSetMock.next()).thenReturn(true);
         when(resultSetMock.getString(NAME_COLUMN)).thenReturn("table");
         when(this.columnMetadataReaderMock.mapColumns("table"))
                 .thenReturn(List.of(ColumnMetadata.builder().name("column").type(DataType.createBool()).build()));
-        final TableMetadataReader metadataReader = createDefaultTableMetadataReader();
+        final TableMetadataReader metadataReader = createTableMetadataReaderWithProperties(
+                new AdapterProperties(Map.of(TableCountLimit.MAXTABLES_PROPERTY, "2000")));
+        final List<String> noFilter = Collections.emptyList();
         final RemoteMetadataReaderException exception = assertThrows(RemoteMetadataReaderException.class,
+                () -> metadataReader.mapTables(resultSetMock, noFilter));
+        assertAll(() -> assertThat(exception.getMessage(), containsString("E-VSCJDBC-42")),
+                () -> assertThat(exception.getMessage(), containsString("2000")));
+    }
+
+    // verify that it does map 3000 tables when the parameter is set so
+    @Test
+    void testValidateMappedTablesListSizeWithProperty3000() throws SQLException {
+        final ResultSet resultSetMock = Mockito.mock(ResultSet.class);
+        // limit number of returned tables, so we can confirm it does map 3000 tables
+        final AtomicInteger tableNumber = new AtomicInteger(0);
```

Comment:
```suggestion
        final AtomicInteger tableCount= new AtomicInteger(0);
```
Also extract configuration of the mock to a method like `resultSetWithUnlimitedSize()`.
Then you can also add a test where the result set contains one table more than the limit.

Reply:
👍